runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ValidateRunnerXlangTest.java

```diff
@@ -221,6 +221,17 @@ protected void pythonDependenciesTest(Pipeline pipeline) {
               .apply(External.of(TEST_PYTHON_BS4_URN, new byte[] {}, expansionAddr));
       PAssert.that(col).containsInAnyOrder("The Dormouse's story");
     }
+
+    protected void combineMultipleTransformTest(Pipeline pipeline) throws IOException {
+      PCollection<String> col =
+          pipeline
+              .apply(Create.of(1L, 2L, 3L))
+              .apply(
+                  External.of(
+                      External.of("map_to_union_types", new byte[] {}, expansionAddr),
```

Comment:
This test will never succeed without the ability to merge consecutive external transforms since `map_to_union_types` return a union of int, string and float types.

Reply:
Reading this test, I wouldn't know where that is defined. Similar for TEST_PREFIX_URN. Let's make this self-contained, e.g. 

```
pipeline
.apply(Create.of(-1, 0, 1))
.apply(PythonMap.viaMapFn("lambda x: 'negative' if x < 0 else x"))
.apply(PythonMap.viaMapFn("type"))
.apply(PythonMap.viaMapFn("str"));
```

which will result in "str", "int", "int".